src/boot/efi/boot.c

```diff
@@ -43,7 +43,8 @@ enum loader_type {
         LOADER_SECURE_BOOT_KEYS,
 };
 
-typedef struct {
+typedef struct ConfigEntry ConfigEntry;
+struct ConfigEntry {
```

Comment:
you could also leave this as it is, and simply make the call() prototype take `const struct ConfigEntry *entry` instead of `const ConfigEntry *entry`.